### PR TITLE
fix(backup uninstall): delete backup having no backup volume manually when uninstall

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: [longhorn/longhorn#5868](https://github.com/longhorn/longhorn/issues/5868)

### CODE REVIEW

File: controller/uninstall_controller.go [hunk 1]
```diff
@@ -617,6 +622,23 @@ func (c *UninstallController) deleteReplicas(replicas map[string]*longhorn.Repli
...
+func (c *UninstallController) deleteLeftBackups(backup *longhorn.Backup) (err error) {
+	backupVolumeName, ok := backup.Labels[types.LonghornLabelBackupVolume]
+	if !ok {
+		// directly delete it if there is even no backup volume label
+		c.ds.DeleteBackup(backup.Name)
+	}
+	_, err = c.ds.GetBackupVolume(backupVolumeName)
+	if err != nil && apierrors.IsNotFound(err) {
+		if err = c.ds.DeleteBackup(backup.Name); err != nil {
+			return errors.Wrap(err, "failed to mark for deletion")
```
Comment (innobead) [inline controller/uninstall_controller.go:635]: Suggest enriching the message, like `failed to delete backup %s`.

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just one comment.

File: controller/uninstall_controller.go [hunk 2]
```diff
@@ -617,6 +622,23 @@ func (c *UninstallController) deleteReplicas(replicas map[string]*longhorn.Repli
 	return
 }
 
+// deleteLeftBackups deletes the backup having no backup volume
+func (c *UninstallController) deleteLeftBackups(backup *longhorn.Backup) (err error) {
+	backupVolumeName, ok := backup.Labels[types.LonghornLabelBackupVolume]
+	if !ok {
+		// directly delete it if there is even no backup volume label
+		c.ds.DeleteBackup(backup.Name)
```
Comment (derekbit) [inline controller/uninstall_controller.go:630]: What happens if this operation fails?

File: controller/uninstall_controller.go [hunk 3]
```diff
@@ -617,6 +622,23 @@ func (c *UninstallController) deleteReplicas(replicas map[string]*longhorn.Repli
...
+	}
+	_, err = c.ds.GetBackupVolume(backupVolumeName)
+	if err != nil && apierrors.IsNotFound(err) {
+		if err = c.ds.DeleteBackup(backup.Name); err != nil {
+			return errors.Wrap(err, "failed to mark for deletion")
+		}
+		return nil
```
Comment (derekbit) [inline controller/uninstall_controller.go:637]: no need to error out when running into a `NotFound` error when deleting a backup.

File: controller/uninstall_controller.go (hunk 2 above)
Comment (ChanYiLin) [reply controller/uninstall_controller.go:630]: Also need to catch the error here, thanks for the remind!

File: controller/uninstall_controller.go (hunk 1 above)
Comment (ChanYiLin) [reply controller/uninstall_controller.go:635]: yeah, I have the same idea, but the whole uninstall process use the same wording...
I will change others to the same wording as well
thanks!

Comment (ChanYiLin) [reply controller/uninstall_controller.go:635]: Oh, I got it, they all have defer function to add more message
Then I will just need to fix this one

Comment (ChanYiLin) [reply controller/uninstall_controller.go]: Fixed

File: controller/uninstall_controller.go (hunk 3 above)
Comment (ChanYiLin) [reply controller/uninstall_controller.go]: Fixed, thanks

File: controller/uninstall_controller.go (hunk 2 above)
Comment (ChanYiLin) [reply controller/uninstall_controller.go]: Fixed.

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@mergify backport v1.5.x v1.4.x v1.3.x

**Pull request merged**